ports/espressif/common-hal/socketpool/Socket.c

```diff
@@ -117,85 +126,86 @@ void socket_user_reset(void) {
         esp_vfs_eventfd_config_t config = ESP_VFS_EVENTD_CONFIG_DEFAULT();
         ESP_ERROR_CHECK(esp_vfs_eventfd_register(&config));
 
-        for (size_t i = 0; i < MP_ARRAY_SIZE(open_socket_fds); i++) {
-            open_socket_fds[i] = -1;
+        // Clear initial socket states
+        for (size_t i = 0; i < MP_ARRAY_SIZE(socket_fd_state); i++) {
+            socket_fd_state[i] = FDSTATE_CLOSED;
             user_socket[i] = NULL;
         }
         socket_change_fd = eventfd(0, 0);
         // Run this at the same priority as CP so that the web workflow background task can be
         // queued while CP is running. Both tasks can still sleep and, therefore, sleep overall.
-        (void)xTaskCreateStaticPinnedToCore(socket_select_task,
+        socket_select_task_handle = xTaskCreateStaticPinnedToCore(socket_select_task,
             "socket_select",
             2 * configMINIMAL_STACK_SIZE,
             NULL,
             uxTaskPriorityGet(NULL),
             socket_select_stack,
-            &socket_select_task_handle,
+            &socket_select_task_buffer,
             xPortGetCoreID());
+    } else {
+        // Not init - close open user sockets
+        for (size_t i = 0; i < MP_ARRAY_SIZE(socket_fd_state); i++) {
+            if ((socket_fd_state[i] == FDSTATE_OPEN) && user_socket[i]) {
+                common_hal_socketpool_socket_close(user_socket[i]);
+            }
+        }
     }
+}
 
-    for (size_t i = 0; i < MP_ARRAY_SIZE(open_socket_fds); i++) {
-        if (open_socket_fds[i] >= 0 && user_socket[i]) {
-            common_hal_socketpool_socket_close(user_socket[i]);
-            int num = open_socket_fds[i];
-            // Close automatically clears socket handle
-            lwip_shutdown(num, SHUT_RDWR);
-            lwip_close(num);
-            open_socket_fds[i] = -1;
-            user_socket[i] = NULL;
-        }
+// Unblock select task (ok if not blocked yet)
+void socketpool_socket_poll_resume(void) {
+    if (socket_select_task_handle) {
+        xTaskNotifyGive(socket_select_task_handle);
     }
 }
 
 // The writes below send an event to the socket select task so that it redoes the
 // select with the new open socket set.
 
 STATIC bool register_open_socket(int fd) {
-    for (size_t i = 0; i < MP_ARRAY_SIZE(open_socket_fds); i++) {
-        if (open_socket_fds[i] == -1) {
-            open_socket_fds[i] = fd;
-            user_socket[i] = false;
-            uint64_t signal = 1;
-            write(socket_change_fd, &signal, sizeof(signal));
-            return true;
-        }
+    if (fd < FD_SETSIZE) {
+        socket_fd_state[fd - LWIP_SOCKET_OFFSET] = FDSTATE_OPEN;
+        user_socket[fd - LWIP_SOCKET_OFFSET] = NULL;
+
+        uint64_t signal = 1;
+        write(socket_change_fd, &signal, sizeof(signal));
+        socketpool_socket_poll_resume();
+        return true;
     }
     return false;
 }
 
+#if 0
+// This is probably not needed
```

Comment:
Go ahead and delete then. `git` has it in the history.

Reply:
Done